lib/stat_tracker.rb

```diff
@@ -92,31 +94,30 @@ def count_of_teams
   end
 
   def winningest_team
-    winningest_team_id = GameTeams.win_percentage.key(GameTeams.win_percentage.values.max)
-    @teams_data.select {|team| team.team_id == winningest_team_id}[0].teamname
+    Team.names_by_id[GameTeams.winningest_team_id]
+  end
+
+  def home_win_percentage
+    GameTeams.win_percentage_hoa("home")
+  end
+
+  def away_win_percentage
+    GameTeams.win_percentage_hoa("away")
   end
 
   def best_fans
-    home_win_percent = GameTeams.win_percentage_hoa("home")
-    away_win_percent = GameTeams.win_percentage_hoa("away")
-    home_win_percent.each do |team, percent|
-      home_win_percent[team] = (percent - away_win_percent[team])
+    home_win_percentage.each do |team, percent|
+      home_win_percentage[team] = (percent - away_win_percentage[team])
```

Comment:
Nice refactor. I think that's a great example of making the code both more condensed & more readable.